protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -45,69 +61,354 @@ pub trait DaSequencerStorage {
 	) -> std::result::Result<(), DaSequencerError>;
 }
 
-pub struct Storage {}
-
 impl Storage {
-	/// Save all batch's Tx in the pending Tx table. The batch's Tx has been verified and validated.
-	pub fn write_batch(
-		&self,
-		batch: DaBatch<FullnodeTx>,
-	) -> std::result::Result<(), DaSequencerError> {
-		todo!();
+	pub fn try_new(path: &str) -> Result<Self, DaSequencerError> {
+		let mut options = Options::default();
+
+		options.create_if_missing(true);
+		options.create_missing_column_families(true);
+
+		let pending_transactions_cf =
+			ColumnFamilyDescriptor::new(cf::PENDING_TRANSACTIONS, Options::default());
+		let blocks_cf = ColumnFamilyDescriptor::new(cf::BLOCKS, Options::default());
+		let blocks_by_digest_cf =
+			ColumnFamilyDescriptor::new(cf::BLOCKS_BY_DIGEST, Options::default());
+
+		let db = DB::open_cf_descriptors(
+			&options,
+			path,
+			[pending_transactions_cf, blocks_cf, blocks_by_digest_cf],
+		)
+		.map_err(|e| DaSequencerError::Generic(e.to_string()))?;
+
+		Ok(Storage { db: Arc::new(db) })
 	}
 
-	/// Return, if exists, the Block at the given height.
-	pub fn get_block_at_height(
-		&self,
-		height: BlockHeight,
-	) -> std::result::Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!();
+	fn determine_next_block_height(&self) -> Result<BlockHeight, DaSequencerError> {
+		let cf = self.db.cf_handle(cf::BLOCKS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing column family: blocks".into())
+		})?;
+
+		let mut iter = self.db.iterator_cf(&cf, rocksdb::IteratorMode::End);
+		if let Some(Ok((key, _value))) = iter.next() {
+			if key.len() != 8 {
+				return Err(DaSequencerError::Generic("Invalid block height key length".into()));
+			}
+
+			let mut arr = [0u8; 8];
+			arr.copy_from_slice(&key);
+			let last_height = u64::from_be_bytes(arr);
+			Ok(BlockHeight(last_height + 1))
+		} else {
+			Ok(BlockHeight(0))
```

Comment:
I think first block height is 1 and it's compatible with the other return path.

Reply:
Why you resolve the conversation with BlockHeight(0)? BlockHeight start at 1?

Reply:
I thought I had committed this, not sure what happened, maybe in the merge conflict something went wrong.